src/PRAS2PowerSystems.jl

```diff
@@ -1,240 +1,265 @@
+    resultsspecs = get_outage_pras_resultspec(DEFAULT_TEMPLATE)
+    results = PRASCore.assess(pras_system, method, resultsspecs...)
+    add_asset_status!(sys, results, DEFAULT_TEMPLATE)
+    return sys
+end
 
-    if (num_scenarios === nothing)
-        num_scenarios = 1
-        @warn "Number of scenarios to be exported not specified. Only exporting one scenario."
-    end
+"""
+    $(TYPEDSIGNATURES)
 
-    # Run PRAS Analysis
-    @info "Running PRAS SequentialMonteCarlo Resource Adequacy analysis for $(num_runs) runs..."
-    shortfall_samples, gens_avail, stors_avail, gen_stors_avail, lines_avail =
-        PRASCore.assess(
-            pras_system,
-            PRASCore.SequentialMonteCarlo(samples=num_runs, threaded=true, verbose=false),
-            PRASCore.ShortfallSamples(),
-            PRASCore.GeneratorAvailability(),
-            PRASCore.StorageAvailability(),
-            PRASCore.GeneratorStorageAvailability(),
-            PRASCore.LineAvailability(),
-        )
-    @info "Successfully completed PRAS Runs. Exporting outage profiles now..."
-    # Setup to save the new PSY System with scenario timeseries data
-    #working_dir = pwd();
-    working_dir = location
-    dt_now = Dates.format(Dates.now(), "dd-u-yy-H-M-S")
-
-    dir_name = joinpath(
-        working_dir,
-        "data",
-        "Generated-Outage-Profile-JSON",
-        string(UUIDs.uuid4()),
-        dt_now,
-    )
+Add the asset status from the worst sample to PSY.TimeSeriesForcedOutage of the component.
+
+# Arguments
+
+  - `sys::PSY.System`: PowerSystems.jl system model
+  - `results::T`: Tuple of results from the PRAS assess call
+  - `template::RATemplate`: PRAS problem template
 
-    #  **TODO: IF generating systems for multiple scenarios, remember to delete the timeseries data from previous scenario.
-    asset_dict = Dict([
-        (:generators, (gens_avail, PSY.Generator)),
-        (:storages, (stors_avail, PSY.Storage)),
-        (:generatorstorages, (gen_stors_avail, PSY.StaticInjection)),
-        (:lines, (lines_avail, PSY.Branch)),
-    ])
-    asset_keys = []
-
-    for k in keys(asset_dict)
-        if (length(getfield(pras_system, k)) != 0)
-            push!(asset_keys, k)
+# Returns
+
+    - PSY System with PSY.TimeSeriesForcedOutage for all components for which asset status is available
+"""
+function add_asset_status!(
+    sys::PSY.System,
+    results::T,
+    template::RATemplate,
+) where {T <: Tuple{Vararg{PRASCore.Results.Result}}}
+    @info ""
+    # Time series timestamps
+    filter_func = x -> (typeof(x) <: PSY.StaticTimeSeries)
+    all_ts = PSY.get_time_series_multiple(sys, filter_func)
+    ts_timestamps = TimeSeries.timestamp(first(all_ts).data)
+
+    shortfall_samp_idx =
+        findfirst(name.(typeof.(results)) .== PRASCore.Results.ShortfallSamplesResult)
+    shortfall_samples = results[shortfall_samp_idx]
+
+    sample_idx = sortperm(shortfall_samples[], rev=true)[1]
```

Comment:
Is there a way to get the maximum index without sorting everything?

Reply:
Done